src/Config/SDK/Configuration/Internal/NodeDefinition/NodeDefinitionTrait.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Config\SDK\Configuration\Internal\NodeDefinition;
+
+trait NodeDefinitionTrait
+{
+    protected bool $allowEmptyValue = true;
+
+    public function cannotBeEmpty(): static
+    {
+        $this->allowEmptyValue = false;
+
+        return $this;
+    }
+
+    public function defaultValue(mixed $value): static
+    {
+        $this->validate()->ifNull()->then(static function () use ($value): mixed {
```

Comment:
this is where I implemented "null/empty values with a default should use the default"